rs/cli/src/store.rs

```diff
@@ -0,0 +1,298 @@
+use std::{io::Read, os::unix::fs::PermissionsExt, path::PathBuf, sync::Arc, time::Duration};
+
+use flate2::bufread::GzDecoder;
+use ic_canisters::governance::governance_canister_version;
+use ic_management_backend::{
+    lazy_registry::{LazyRegistry, LazyRegistryImpl},
+    proposal::ProposalAgent,
+    registry::sync_local_store_with_path,
+};
+use ic_management_types::Network;
+use ic_registry_local_registry::LocalRegistry;
+use log::{debug, info, warn};
+
+use crate::{
+    auth::Neuron,
+    commands::IcAdminVersion,
+    cordoned_feature_fetcher::{CordonedFeatureFetcher, CordonedFeatureFetcherImpl},
+    ic_admin::{IcAdmin, IcAdminImpl},
+};
+
+#[derive(Clone)]
+pub struct Store {
+    path: PathBuf,
+    offline: bool,
+}
+
+const DURATION_BETWEEN_CHECKS_FOR_NEW_IC_ADMIN: Duration = Duration::from_secs(60 * 60 * 24);
+pub const FALLBACK_IC_ADMIN_VERSION: &str = "d4ee25b0865e89d3eaac13a60f0016d5e3296b31";
+
+impl Store {
+    #[cfg(not(test))]
+    pub fn new(offline: bool) -> anyhow::Result<Self> {
+        Self::new_inner(offline, "dre-store")
+    }
+
+    // Really important to distinguish from test and
+    // real store because test store, if not handled
+    // correctly, can leave an invalid state
+    #[cfg(test)]
+    pub fn new(offline: bool) -> anyhow::Result<Self> {
+        Self::new_inner(offline, "dre-test-store")
+    }
+
+    fn new_inner(offline: bool, store_name: &str) -> anyhow::Result<Self> {
+        Ok(Self {
+            path: dirs::cache_dir()
+                .ok_or(anyhow::anyhow!("Couldn't find cache dir for dre store"))?
+                .join(store_name),
+            offline,
+        })
+    }
+
+    pub fn is_offline(&self) -> bool {
+        self.offline
+    }
+
+    fn path(&self) -> &PathBuf {
+        &self.path
+    }
+
+    fn local_store_for_network(&self, network: &Network) -> anyhow::Result<PathBuf> {
+        let local_store_dir = self.path().join("local_registry").join(&network.name);
+
+        if !local_store_dir.exists() {
+            debug!(
+                "Directory for local store for network {} doesn't exist. Creating on path `{}`",
+                network.name,
+                local_store_dir.display()
+            );
+            std::fs::create_dir_all(&local_store_dir)?
+        }
+
+        Ok(local_store_dir)
+    }
+
+    fn guest_labels_cache_dir(&self, network: &Network) -> anyhow::Result<PathBuf> {
+        let dir = self.path().join("labels").join(&network.name);
+
+        if !dir.exists() {
+            debug!(
+                "Directory for labels for network {} doesn't exist. Creating on path `{}`",
+                network.name,
+                dir.display()
+            );
+            std::fs::create_dir_all(&dir)?
+        }
+
+        Ok(dir)
+    }
+
+    #[cfg(test)]
+    pub fn guest_labels_cache_path_outter(&self, network: &Network) -> anyhow::Result<PathBuf> {
```

Comment:
```suggestion
    pub fn guest_labels_cache_path_outer(&self, network: &Network) -> anyhow::Result<PathBuf> {
```

outter ==> outer

although, having a "test" in the name is clearer for me, since the purpose is clearer.